### PR TITLE
[Refactor][RayCluster] Replace RayClusterReconciler.Log with LogConstructor

### PR DESCRIPTION
**Pull request by rueian**
## Why are these changes needed?

As @astefanutti suggested here https://github.com/ray-project/kuberay/pull/1945#discussion_r1503835776, it could be better to use the `LogConstructor` option to construct a logger because the logger is shared between controller-runtime and the controller itself.

But I then gave the wrong [example](https://github.com/ray-project/kuberay/pull/1945#discussion_r1504214798). Now I would like to fix that.

This PR uses the `LogConstructor` inside the `WithOption` call instead of a `WithLogConstructor`, which will be overridden by the former one, and then removes the no longer needed`RayClusterReconciler.Log`.

The `r.Log` loggers in the tests are also removed since they were already not used in tests previously.

## Screenshot

This is what the logs look like, including the reconcileID.

image

## Related issue number

https://github.com/ray-project/kuberay/issues/1947

## Checks

- [x] I've made sure the tests are passing. 
- Testing Strategy
   - [x] Unit tests
   - [x] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by astefanutti — APPROVED** [review]
Nice. Maybe that can be done for the other controllers as well.

File: ray-operator/controllers/ray/rayservice_controller.go [hunk 1]
```diff
@@ -338,6 +337,15 @@ func (r *RayServiceReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Owns(&rayv1.RayCluster{}).
 		Owns(&corev1.Service{}).
 		Owns(&networkingv1.Ingress{}).
+		WithOptions(controller.Options{
+			LogConstructor: func(request *reconcile.Request) logr.Logger {
+				logger := ctrl.Log.WithName("controllers").WithName("RayJob")
+				if request != nil {
+					logger = logger.WithValues("RayJob", request.NamespacedName)
```
Comment (astefanutti) [inline ray-operator/controllers/ray/rayservice_controller.go:344]: I think that's `RayService` here.

Comment (rueian) [reply ray-operator/controllers/ray/rayservice_controller.go]: Oops, indeed. Thanks!

**Review by astefanutti — APPROVED** [review]
LGTM

**Review by kevin85421 — APPROVED** [review]

**Comment by kevin85421** [comment]
Thanks, @rueian and @astefanutti!

**Pull request merged**